ledger/metrics/src/main/scala/com/daml/metrics/Metrics.scala

```diff
@@ -273,12 +274,59 @@ class Metrics(val registry: MetricRegistry) {
         val stopDeduplicatingCommand: Timer =
           registry.timer(prefix :+ "stop_deduplicating_command")
 
-        def deserialization(description: String): Timer =
-          registry.timer(prefix :+ description :+ "deserialization")
-        def wait(description: String): Timer = registry.timer(prefix :+ description :+ "wait")
-        def exec(description: String): Timer = registry.timer(prefix :+ description :+ "exec")
-        val waitAll: Timer = wait("all")
-        val execAll: Timer = exec("all")
+        private val createDatabaseMetrics: String => DatabaseMetrics =
+          DatabaseMetrics(registry, prefix)(_)
+
+        private val overall = createDatabaseMetrics("all")
```

Comment:
It'd be great if we could come up with an API that allows for overall tracking of deserialization too. For example, passing the deserializer to the `DbDispatcher` as an optional second parameter.

Reply:
I understand the point and I see the advantage, but this would require the `DbDispatcher` to be aware of the output and use deserialization in some non-dumb way. I'm afraid this would make `DbDispatcher` cover too many concerns. It sounds like probably we could do a better job at pulling apart `executeSql` in two parts (one that handles wait and the other that handles execution) and than we have a single place where we compose the two, possibly with the addition of deserization. I'll take this as an improvement for the future, thanks for raising this. :slightly_smiling_face: